### ISSUE TITLE
Not able to resume experiments from checkpoints in the PTL dev version

### ISSUE THREAD

**Issue by MartaTintore** [PyTorchLightning/pytorch-lightning#9355]
## 🐛 Bug

I am using the dev version of PTL (installed PTL from source) to avoid seeing the training loss being added instead of averaged (a bug that is reported here https://github.com/PyTorchLightning/pytorch-lightning/pull/9142 and solved in the dev version) but with the dev version I am not able to resume experiments from checkpoints anymore... I pass the checkpoint file but it doesnt seem to care about it. Resuming from checkpoint was working fine in version 1.4.2.

### Environment

- PyTorch Lightning Version (e.g., 1.3.0): 1.5.0.dev0
- PyTorch Version (e.g., 1.8) 1.9.0
- Python version: 3.8.5
- Cudatoolkit: 10.2.89
- How you installed PyTorch (`conda`, `pip`, source): conda
- If compiling from source, the output of `torch.__config__.show()`: 

`'PyTorch built with:\n  - GCC 7.3\n  - C++ Version: 201402\n  - Intel(R) Math Kernel Library Version 2020.0.0 Product Build 20191122 for Intel(R) 64 architecture applications\n  - Intel(R) MKL-DNN v2.1.2 (Git Hash 98be7e8afa711dc9b66c8ff3504129cb82013cdb)\n  - OpenMP 201511 (a.k.a. OpenMP 4.5)\n  - NNPACK is enabled\n  - CPU capability usage: AVX2\n  - CUDA Runtime 10.2\n  - NVCC architecture flags: -gencode;arch=compute_37,code=sm_37;-gencode;arch=compute_50,code=sm_50;-gencode;arch=compute_60,code=sm_60;-gencode;arch=compute_70,code=sm_70\n  - CuDNN 7.6.5\n  - Magma 2.5.2\n  - Build settings: BLAS_INFO=mkl, BUILD_TYPE=Release, CUDA_VERSION=10.2, CUDNN_VERSION=7.6.5, CXX_COMPILER=/opt/rh/devtoolset-7/root/usr/bin/c++, CXX_FLAGS= -Wno-deprecated -fvisibility-inlines-hidden -DUSE_PTHREADPOOL -fopenmp -DNDEBUG -DUSE_KINETO -DUSE_FBGEMM -DUSE_QNNPACK -DUSE_PYTORCH_QNNPACK -DUSE_XNNPACK -DSYMBOLICATE_MOBILE_DEBUG_HANDLE -O2 -fPIC -Wno-narrowing -Wall -Wextra -Werror=return-type -Wno-missing-field-initializers -Wno-type-limits -Wno-array-bounds -Wno-unknown-pragmas -Wno-sign-compare -Wno-unused-parameter -Wno-unused-variable -Wno-unused-function -Wno-unused-result -Wno-unused-local-typedefs -Wno-strict-overflow -Wno-strict-aliasing -Wno-error=deprecated-declarations -Wno-stringop-overflow -Wno-psabi -Wno-error=pedantic -Wno-error=redundant-decls -Wno-error=old-style-cast -fdiagnostics-color=always -faligned-new -Wno-unused-but-set-variable -Wno-maybe-uninitialized -fno-math-errno -fno-trapping-math -Werror=format -Wno-stringop-overflow, LAPACK_INFO=mkl, PERF_WITH_AVX=1, PERF_WITH_AVX2=1, PERF_WITH_AVX512=1, TORCH_VERSION=1.9.0, USE_CUDA=ON, USE_CUDNN=ON, USE_EXCEPTION_PTR=1, USE_GFLAGS=OFF, USE_GLOG=OFF, USE_MKL=ON, USE_MKLDNN=ON, USE_MPI=OFF, USE_NCCL=ON, USE_NNPACK=ON, USE_OPENMP=ON, \n'`

**Comment by justusschock** [comment]
Hi @MartaTintore 

I would like to look into this. Do you have a minimal example with which I can reproduce this?

Best, 
Justus

**Comment by MartaTintore** [comment]
I am using this function: https://github.com/CompVis/taming-transformers/blob/master/main.py

**Comment by justusschock** [comment]
Hi @MartaTintore ,

I looked into this and unfortunately was not able to reproduce it there. Could you try to put together a minimal example using the `BoringModel` and some random data? 

Thanks
